testing/test_float4.cpp

```diff
@@ -1,33 +1,81 @@
 #include "testing.h"
 #include <tack/float4.h>
+#include <array>
 
-struct test_float4_t : public test_t {
+struct test_float4_extract_t : public test_t {
 
-    test_float4_t()
-        : test_t("test_float4")
+    test_float4_extract_t()
+        : test_t("test_float4_extract")
     {
         register_test(this);
     }
 
-    void test1()
+    virtual bool run() override
     {
         using namespace tack;
+        const float4 a(1.f, 2.f, 3.f, 4.f);
+        pass_ &= a.get<e_x>() == 1.f;
+        pass_ &= a.get<e_y>() == 2.f;
```

Comment:
I find `pass_ &=` hard to read, I have thoughts on this though which you will see soon.